JamJar/JamJarPageViewController.swift

```diff
@@ -0,0 +1,371 @@
+    @IBOutlet weak var likesCountLabel: UILabel!
+    @IBOutlet weak var dislikesCountLabel: UILabel!
+    @IBOutlet weak var flagButton: UIButton!
+    @IBOutlet weak var likeButton: UIButton!
+    @IBOutlet weak var dislikeButton: UIButton!
+    
+
+    //Variable to store video frame
+    var jamJarContainerFrameInPortrait: CGRect!
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        // Do any additional setup after loading the view, typically from a nib.
+        
+        //Save the potrait jamjar frame
+        jamJarContainerFrameInPortrait = self.jamJarContainerView.frame
+        
+        //Add action for concert tapped
+        let concertTap = UITapGestureRecognizer(target: self, action: #selector(VideoPageViewController.concertTapped(_:)))
+        self.concertInfoView.addGestureRecognizer(concertTap)
+        
+        // Register the reusable video cell
+        self.suggestedTableView.registerNib(UINib(nibName: "JamJarCell", bundle: nil), forCellReuseIdentifier: "JamJarCell")
+        self.suggestedTableView.registerNib(UINib(nibName: "JamJarHeader", bundle: nil), forCellReuseIdentifier: "JamJarHeader")
+        
+        // Set Report Reasons Data
+        self.reasonsToFlag = ["Accuracy", "Inappropriate", "Quality","Report User"];
+        
+        self.suggestedTableView.reloadData()
+    }
+    
+    override func didReceiveMemoryWarning() {
+        super.didReceiveMemoryWarning()
+        // Dispose of any resources that can be recreated.
+    }
+    
+    deinit {
+        //When the video page is dismissed, remove observers from the AVPlayerController
+        removeOberservsInPlayer()
+    }
+    
+    // Record a view
+    func recordView(videoId: Int!) {
+        VideoService.watchingVideo(videoId)
+    }
+    
+    //make embedded controller full screen
+    func fullScreenVideo() {
+        //hide navigation bar and tool bar
+        self.navigationController?.navigationBar.hidden = true
+        self.tabBarController?.tabBar.hidden = true
+        
+        UIView.animateWithDuration(0.25) {
+            self.jamJarContainerView.frame = CGRect(x: 0, y: 0, width: self.view.frame.width, height: self.view.frame.height)
+        }
+    }
+    
+    //Worst function name ever
+    func unfullScreenVideo() {
+        //show navigation bar and tool bar
+        self.navigationController?.navigationBar.hidden = false
+        self.tabBarController?.tabBar.hidden = false
+        
+        UIView.animateWithDuration(0.25) {
+            self.jamJarContainerView.frame = self.jamJarContainerFrameInPortrait
+        }
+        
+        self.view.layoutIfNeeded()
+    }
+    
+    func removeOberservsInPlayer() {
+        for controller in self.childViewControllers {
+            if let child = controller as? StitchedJamJarAVPlayerViewController {
+                child.removeObservers()
+            }
+        }
+    }
+    
+    func changeJamJarInPlayer() {
```

Comment:
What does this function do?  Does it hot-swap a jamjar?  Does it change the video?  I'm confused :/